server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -199,6 +204,10 @@ Metadata upgradeMetadataForNode(Metadata metadata,
         return upgradeMetadata(metadata, metadataIndexUpgradeService, metadataUpgrader);
     }
 
+    Metadata bugFixMetadataForNode(Metadata metadata) {
+        return Bug13380.fix(metadata);
+    }
+
```

Comment:
Wouldn't it be possible to use the existing plugging mechanism (`MetadataUpgrader`) to do this ?

Reply:
It requires `UnaryOperator<Metadata>` so the existing mechanism needs to be expanded.